packages/core/logger/src/index.js

```diff
@@ -33,137 +31,98 @@ export class Logger {
       }));
     };
     this.logFile = options.logFile;
-    this.context = options.context;
   }
 
-  /**
-   * Parses the logFile, returning an array of JSON objects containing the
-   * log messages.
-   * @param {Number} limit specifies how many log messages to return from the
-   *  end of the log file
-   * @returns {Array} array containing
-   *  - msg: the log message
-   *  - logLevel: log level (ie 'info', 'debug')
-   *  - name: process name (always "embark")
-   *  - timestamp: timestamp of log message (milliseconds since 1/1/1970)
-   */
-  parseLogFile(limit) {
-    let matches;
-    let logs = [];
-    const logFile = fs.readFileSync(this.logFile, 'utf8');
-    while ((matches = LOG_REGEX.exec(logFile)) !== null) {
-     // This is necessary to avoid infinite loops with zero-width matches
-     if (matches.index === LOG_REGEX.lastIndex) {
-       LOG_REGEX.lastIndex++;
-     }
-
-     if (matches && matches.length) {
-       logs.push({
-         msg: [matches[3]],
-         logLevel: matches[2],
-         name: 'embark',
-         timestamp: date.parse(matches[1], DATE_FORMAT).getTime()
-       });
-     }
-    }
 
-    // if 'limit' is specified, get log lines from the end of the log file
-    if(limit && limit > 0 && logs.length > limit){
-      logs.slice(limit * -1);
-    }
-    return logs;
+  registerAPICall(plugins) {
+    let plugin = plugins.createPlugin('dashboard', {});
+    plugin.registerAPICall(
+      'ws',
+      '/embark-api/logs',
+      (ws, _req) => {
+        this.events.on("log", (logLevel, logMsg) => {
+          logMsg = escapeHtml(logMsg);
+          ws.send(JSON.stringify({msg: logMsg, msg_clear: logMsg.stripColors, logLevel: logLevel}), () => {});
+        });
+      }
+    );
   }
-}
 
-Logger.prototype.registerAPICall = function (plugins) {
-  const self = this;
-
-  let plugin = plugins.createPlugin('dashboard', {});
-  plugin.registerAPICall(
-    'ws',
-    '/embark-api/logs',
-    (ws, _req) => {
-      self.events.on("log", function (logLevel, logMsg) {
-        logMsg = escapeHtml(logMsg);
-        ws.send(JSON.stringify({msg: logMsg, msg_clear: logMsg.stripColors, logLevel: logLevel}), () => {});
-      });
+  writeToFile(_txt) {
+    if (!this.logFile) {
+      return;
     }
-  );
-};
 
-Logger.prototype.writeToFile = function (_txt) {
-  if (!this.logFile) {
-    return;
-  }
-
-  let origin = "[" + ((new Error().stack).split("at ")[3]).trim() + "]";
+    let origin = "[" + ((new Error().stack).split("at ")[3]).trim() + "]";
 
-  const formattedDate = [`[${date.format(new Date(), DATE_FORMAT)}]`]; // adds a timestamp to the logs in the logFile
-  fs.appendFileSync(this.logFile, "\n" + formattedDate.concat(origin, Array.from(arguments)).join(' '));
-};
+    const formattedDate = [`[${date.format(new Date(), DATE_FORMAT)}]`]; // adds a timestamp to the logs in the logFile
+    fs.appendFileSync(this.logFile, "\n" + formattedDate.concat(origin, Array.from(arguments)).join(' '));
```

Comment:
It would be nice if we could switch to using the async versions of the `fs` methods, if it's feasible to do so.

That would make it a little more difficult to test whether the logger writes as expected, since the methods (e.g. `logger.info`) are fire-and-forget; however, in the test you could use `fs.fstatSync` before calling a logger method and then use `fs.fstatSync` in a setTimeout loop to compare the `mtime` — once it's changed you know you can check what was written to the log file.

Reply:
It just occurred to me that if we did switch to async `fs` methods we'd need to serialize the writes using an `async.cargo` (or some [roll-your-own async queue](https://codesandbox.io/s/async-night-e70mf)) like we do in the transaction logger.

Reply:
Or the write could be `await`ed instead of having it be fire and forget.

Reply:
I think using `await` could still result in overlapping write-attempts with `fs.appendFile` since the `await` pertains only within a call of the async function, i.e. it won't block multiple calls of that function.

Reply:
That's a good point, maybe a queue would be best then.

Reply:
Is there overall consensus that we should refactor this to async APIs? 
Happy to do it if there are strong opinions.